programs/lfx-mentorship/2023/03-Sep-Nov/project_ideas.md

```diff
@@ -30,6 +30,27 @@
   - Kevin Hannon, @kannon92, kevin@gr-oss.io
 - Upstream Issue (URL): https://github.com/armadaproject/armada/issues/2702
 
+### CRI-O
+
+#### Add additional log drivers to conmon-rs
+
+- Description: conmon-rs is a container monitor written in Rust, used by CRI-O to monitor a container's lifecycle. Part of its responsibilities is log forwarding--taking the output of the container and writing that output to various places. Currently, conmon-rs supports one format: the one required by the Kubernetes CRI. The goal of this proposal is to add new log formats from the list of standardized ones, like JSON, Splunk, Journald.
+- Expected outcome: A JSON log driver and Journald log driver are added to conmon-rs. A stretch goal of adding a Splunk log driver is also within scope.
+- Recommended Skills: Rust, familiarity with containers
+- Mentor(s):
+  - Peter Hunt, haircommander, pehunt@redhat.com
+  - Sascha Grunert, saschagrunert, sgrunert@redhat.com
+- Upstream Issue (URL): https://github.com/containers/conmon-rs/issues/1126
+
+#### Automate CRI-O release process
+
+- Description: CRI-O packages to openSUSE's [OBS](build.opensuse.org) to provide packages for a variety of Linux distributions. Much of this process is manual today, and could be automated.
+- Expected outcome: A github action that reacts to a tag being pushed to CRI-O that packages CRI-O to all the required distributions.
+- Recommended Skills: Golang
+- Mentor(s):
+  - Peter Hunt, haircommander, pehunt@redhat.com
+- Upstream Issue (URL): https://github.com/cri-o/cri-o/issues/4003
```

Comment:
I'm not sure we should add that. We could wait for the official Kubernetes OBS support and integrate into it.

See KEP: https://github.com/kubernetes/enhancements/issues/1731 
Or board: https://github.com/orgs/kubernetes/projects/137/views/1

Reply:
I assume we'll have something concrete by the end of the year, the alpha OBS implementation will be done with v1.28.

Reply:
Do you need to make an update to this before I merge it in?

Reply:
We can also leave it, just wanted to highlight overlapping efforts here.

Reply:
I agree, let's drop it for now and maybe consider term 4 once upstream has fully committed to a direction